include/ros_control_boilerplate/generic_hw_interface.h

```diff
@@ -80,9 +80,23 @@ class GenericHWInterface : public hardware_interface::RobotHW
   /** \brief Read the state from the robot hardware. */
   virtual void read(ros::Duration &elapsed_time) = 0;
 
+  /** \brief Delegate RobotHW::read() calls to read() */
+  virtual void read(const ros::Time& /*time*/, const ros::Duration& period)
+  {
+    ros::Duration elapsed_time = period;
+    read(elapsed_time);
+  }
+
   /** \brief Write the command to the robot hardware. */
   virtual void write(ros::Duration &elapsed_time) = 0;
 
+  /** \brief Delegate RobotHW::write() calls to write() */
+  virtual void write(const ros::Time& /*time*/, const ros::Duration& period)
```

Comment:
why is the first argument necessary? Since I don't see it used in the function.

Reply:
OK, I understand now. The first argument is there because you overloaded the RobotHW function and wanted to avoid a compiler warning. Well, that's kind of messy but I guess there's nothing to be done about it.

Reply:
Hey, thats exactly it. As this project uses c++11 i can add an ```override``` specifier to make this clearer.